chunk/chunk_store.go

```diff
@@ -297,77 +376,84 @@ func (c *Store) lookupMatchers(ctx context.Context, from, through model.Time, ma
 		}
 	}
 
+	// Merge chunkSets in order
```

Comment:
Not sure this comment is helpful. More interesting to me would be a comment saying why this is an intersect and not a merge.